scalableminds/webknossos
# Improve signup flow

~~@fm3 Could you add a boolean `enableAutoVerify` property to organizations which can be set in the DB.
I'll take care of using this in the frontend to auto-login after user registration if it is enabled :)~~ Thanks :)

~~@philippotto Not ready for review yet, but maybe you have some feedback when looking at this gif :)~~

![signup-flow](https://user-images.githubusercontent.com/1702075/52355746-39077700-2a33-11e9-995a-40f4bfdd9b9f.gif)

> As you can see, I added a modal to allow for user registration/login with minimal disruption.
The default org is prefilled.
As autoVerify is not yet implemented, I used another user to login after registration (with autoVerify enabled, the user would be logged in automatically). Ignore the accidental rightclick ^^
Also note that the dataset position and zoom is transferred to the created tracing so that users ends up in the same spot they looked at before :)
Any thoughts on that already?

If the user clicks on the `Register Now` link in the navbar and registers successfully, EITHER (if autoVerify is enabled for the user's organization), the user is logged in automatically and goBack() is invoked, so the user returns to the last page, OR (if autoVerify is not enabled) the user is forwarded to the login page.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Check out locally, so you can modify the DB and application.conf values.
- Fill in `defaultOrganization = "Connectomics_Department"` in the application.conf and visit the register page. The organization should be prefilled.
  - Supplying the organization via URL should still have precedence (`?organizationName=...`)
- Logout and visit the spotlight page. Basically replicate the signup flow as depicted in the GIF (once with `enableAutoVerify` off and on).
  - enable using `update webknossos.organizations set enableAutoVerify = true where name = 'Connectomics_Department'`
  - If `enableAutoVerify` is on, the user should be logged in automatically after registering in the modal and the tracing should be created.
  - The created tracing should honor the position and zoomstep of the viewed dataset, so the user is in the same spot as when clicking `Create Tracing`
- Not sure whether the discuss authentication can be tested on a dev server, ideally we should test this before merging

### Issues:
- fixes #3683 
- fixes #3745

------
~~- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
